### PR TITLE
Fix: Python SDK: kustomize.Directory: handle a not fully configured provider

### LINKED ISSUE #2346: pulumi_kubernetes.kustomize.Directory blows up in Python when the cluster does not exist yet
### What happened?

I am trying to write a Python Pulumi program that will create an AWS EKS cluster and deploy a kustomization in it.

The Pulumi program from "Steps to reproduce" results in:

```console
pulumi_kubernetes_issue $ pulumi preview                          
Previewing update (dev):
     Type                                   Name                                    Plan       Info
 +   pulumi:pulumi:Stack                    pulumi_kubernetes_issue-dev             create     1 error
 +   ├─ eks:index:Cluster                   test                                    create     
 +   │  ├─ eks:index:ServiceRole            test-eksRole                            create     
 +   │  │  ├─ aws:iam:Role                  test-eksRole-role                       create     
 +   │  │  └─ aws:iam:RolePolicyAttachment  test-eksRole-4b490823                   create     
 +   │  ├─ eks:index:ServiceRole            test-instanceRole                       create     
 +   │  │  ├─ aws:iam:Role                  test-instanceRole-role                  create     
 +   │  │  ├─ aws:iam:RolePolicyAttachment  test-instanceRole-3eb088f2              create     
 +   │  │  ├─ aws:iam:RolePolicyAttachment  test-instanceRole-03516f97              create     
 +   │  │  └─ aws:iam:RolePolicyAttachment  test-instanceRole-e1b295bd              create     
 +   │  ├─ eks:index:RandomSuffix           test-cfnStackName                       create     
 +   │  ├─ aws:iam:InstanceProfile          test-instanceProfile                    create     
 +   │  ├─ aws:ec2:SecurityGroup            test-eksClusterSecurityGroup            create     
 +   │  ├─ aws:ec2:SecurityGroupRule        test-eksClusterInternetEgressRule       create     
 +   │  ├─ aws:eks:Cluster                  test-eksCluster                         create     
 +   │  ├─ pulumi:providers:kubernetes      test-eks-k8s                            create     
 +   │  ├─ eks:index:VpcCni                 test-vpc-cni                            create     
 +   │  ├─ aws:ec2:SecurityGroup            test-nodeSecurityGroup                  create     
 +   │  ├─ kubernetes:core/v1:ConfigMap     test-nodeAccess                         create     
 +   │  ├─ aws:ec2:SecurityGroupRule        test-eksNodeInternetEgressRule          create     
 +   │  ├─ aws:ec2:SecurityGroupRule        test-eksNodeIngressRule                 create     
 +   │  ├─ aws:ec2:SecurityGroupRule        test-eksExtApiServerClusterIngressRule  create     
 +   │  ├─ aws:ec2:SecurityGroupRule        test-eksClusterIngressRule              create     
 +   │  ├─ aws:ec2:SecurityGroupRule        test-eksNodeClusterIngressRule          create     
 +   │  ├─ aws:ec2:LaunchConfiguration      test-nodeLaunchConfiguration            create     
 +   │  ├─ aws:cloudformation:Stack         test-nodes                              create     
 +   │  └─ pulumi:providers:kubernetes      test-provider                           create     
 +   ├─ pulumi:providers:kubernetes         test                                    create     
 +   └─ kubernetes:kustomize:Directory      test                                    create     


Diagnostics:
  pulumi:pulumi:Stack (pulumi_kubernetes_issue-dev):
    error: Program failed with an unhandled exception:
    Traceback (most recent call last):
      File "/Users/Filip.Zyzniewski/jira/DOD-3178/pulumi_kubernetes_issue/./__main__.py", line 14, in <module>
        pulumi_kubernetes.kustomize.Directory(
      File "/Users/Filip.Zyzniewski/jira/DOD-3178/pulumi_kubernetes_issue/venv/lib/python3.10/site-packages/pulumi_kubernetes/kustomize/kustomize.py", line 117, in __init__
        __ret__ = pulumi.runtime.invoke(
    TypeError: 'NoneType' object is not subscriptable


pulumi_kubernetes_issue $
```

### Expected Behavior

A successful pulumi preview.

### Steps to reproduce

1. Create a Pulumi project with the following structure:

`Pulumi.yaml`:

```yaml
name: pulumi_kubernetes_issue
runtime:
  name: python
  options:
    virtualenv: venv
description: Reproducing a kubernetes.kustomize.Directory issue
```

`Pulumi.dev.yaml`:

```yaml
encryptionsalt: v1:5oLFaR1GrWo=:v1:8bU7V4q6vDz01C28:pi6s9hrBI+Y9ADwrH3K8dM3muyiiJQ==
config:
  aws:region: eu-west-1
```

`__main__.py`:

```python
import pulumi
import pulumi_eks
import pulumi_kubernetes

cluster = pulumi_eks.Cluster("test")

provider = pulumi_kubernetes.Provider(
    "test",
    kubeconfig=cluster.kubeconfig.apply(
        lambda config: json.dumps(config, sort_keys=True)
    ),
)

pulumi_kubernetes.kustomize.Directory(
    "test",
    directory="./directory",
    opts=pulumi.ResourceOptions(provider=provider),
)
```

`directory/kustomization.yaml`: empty

`requirements.txt`:
```ini
pulumi==3.60.1
pulumi-eks==1.0.1
pulumi-kubernetes==3.24.2
```

2. Run:

```console
pulumi_kubernetes_issue $ python3 -m venv venv
pulumi_kubernetes_issue $ venv/bin/pip install -r requirements.txt
Requirement already satisfied: pulumi==3.60.1 in ./venv/lib/python3.10/site-packages (from -r requirements.txt (line 1)) (3.60.1)
[...]
Successfully installed certifi-2022.12.7 charset-normalizer-3.1.0 idna-3.4 pulumi-eks-1.0.1 pulumi-kubernetes-3.24.2 requests-2.28.2 urllib3-1.26.15
pulumi_kubernetes_issue $ pulumi preview 
```

### Output of `pulumi about`

```console
pulumi_kubernetes_issue $ pulumi about
CLI          
Version      3.60.1
Go Version   go1.20.2
Go Compiler  gc

Plugins
NAME        VERSION
aws         5.34.0
eks         1.0.1
kubernetes  3.24.2
python      unknown

Host     
OS       darwin
Version  13.2
Arch     arm64

This project is written in python: executable='/opt/homebrew/bin/python3' version='3.10.9
'

Backend        
Name           JFYYQW9DJ7
URL            file://~
User           Filip.Zyzniewski
Organizations  

Dependencies:
NAME        VERSION
pip         23.0.1
pulumi-eks  1.0.1
setuptools  67.6.1
wheel       0.40.0

Pulumi locates its logs in /var/folders/xc/rb73zyjd3kq26tv0hpk4wxh40000gq/T/ by default
warning: Failed to get information about the current stack: No current snapshot
pulumi_kubernetes_issue $
```

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by filip-zyzniewski**
### Proposed changes

When the kubernetes provider is not fully configured (https://github.com/pulumi/pulumi/blob/v3.60.1/sdk/go/common/resource/plugin/provider_plugin.go#L1364-L1367), then the `value` property of the `InvokeResult` instance is `None`.

This PR attempts to add handling of this case.

### Related issues (optional)

Fixes #2346

### CODE REVIEW

File: sdk/python/pulumi_kubernetes/kustomize/kustomize.py [hunk 1]
```diff
@@ -115,12 +115,16 @@ def omit_resource(obj, opts):
                                            provider=opts.provider if opts.provider else None)
 
         __ret__ = pulumi.runtime.invoke(
-            'kubernetes:kustomize:directory', {'directory': directory}, invoke_opts).value['result']
+            'kubernetes:kustomize:directory', {'directory': directory}, invoke_opts)
+
+        # Handle the cases when the provider is not fully configured:
+        #   https://github.com/pulumi/pulumi/blob/v3.60.1/sdk/go/common/resource/plugin/provider_plugin.go#L1364-L1367
+        result = (__ret__.value or {}).get('result', [])
```
Comment (filip-zyzniewski) [inline sdk/python/pulumi_kubernetes/kustomize/kustomize.py:122]: Alternatively: 

```suggestion
        result = (__ret__.value or {'result': []})['result']
```

**Comment by kpitzen** [comment]
Hi @filip-zyzniewski - thank you so much for your contribution here! It looks like this fix was implemented directly in the Python SDK. Those files are autogenerated from two sources - one is the K8s API spec, and the other is from overlays defined in this repo. If you could re-implement this fix in this file here:
https://github.com/pulumi/pulumi-kubernetes/blob/master/provider/pkg/gen/python-templates/kustomize/kustomize.py
and then run `make python_sdk`, I think we'll be in a good place to run tests here.
Thanks again!

**Comment by filip-zyzniewski** [comment]
> If you could re-implement this fix in this file here: https://github.com/pulumi/pulumi-kubernetes/blob/master/provider/pkg/gen/python-templates/kustomize/kustomize.py and then run `make python_sdk`, I think we'll be in a good place to run tests here.

Hi @kpitzen, thanks, done in f8cf6b33aa89c9db06d52c2e70af29fe7d059991 , PTAL.

**Comment by kpitzen** [comment]
Awesome @filip-zyzniewski - this is looking great! One final ask - would you mind adding an entry to `CHANGELOG.md` under the `Unreleased` section describing this fix? Tests are passing and it seems like this will accomplish what we'd hope.

Thank you again!

**Comment by filip-zyzniewski** [comment]
> One final ask - would you mind adding an entry to `CHANGELOG.md` under the `Unreleased` section describing this fix? 

Thanks @kpitzen, done in a136e43922258bbb9d1cff66035e15c8d612f3c5, PTAL.

**Comment by kpitzen** [comment]
@filip-zyzniewski - the change looks good, all tests are passing, and we have a changelog entry! ✅ 

Going to get this merged now, and it'll be available in our next release.

**Pull request merged**